packages/valory/skills/decision_maker_abci/behaviours/decision_request.py

```diff
@@ -139,240 +66,18 @@ def setup(self) -> None:
         )
         prompt = self.params.prompt_template.substitute(prompt_params)
         tool = self.synchronized_data.mech_tool
-        self._metadata = MechMetadata(prompt, tool)
+        nonce = str(uuid4())
+        self._metadata = MechMetadata(prompt, tool, nonce)
```

Comment:
This was not removed:

https://github.com/valory-xyz/trader/blob/ac8e72dc1124ec6f8c264dc9d9910603615d274b/packages/valory/skills/decision_maker_abci/behaviours/decision_request.py#L64-L70

Also why is the nonce not generated automatically anymore?

Reply:
The nonce was generated on the class MechMetadata (Trader): https://github.com/valory-xyz/trader/blob/ac8e72dc1124ec6f8c264dc9d9910603615d274b/packages/valory/skills/decision_maker_abci/behaviours/decision_request.py#L70

This class is now imported from the mech_interact_abci, and removed the code duplication. The class from mech_interact_abci does not automatically generate the nonce: https://github.com/valory-xyz/IEKit/blob/8be2ce6acecf80030e6cd66b122da64cc524d57d/packages/valory/skills/mech_interact_abci/states/base.py#L51

If this is not the intended behaviour, let me know and I'll port the functionality to mech_interact_abci.

Reply:
Added issue on IEKit repository to address it https://github.com/valory-xyz/IEKit/issues/156